packages/lib/gpcircuits/circuits/proto-pod-gpc.circom

```diff
@@ -108,9 +109,61 @@ template ProtoPODGPC (
         entryValueHashes[i] <== entryProofSiblings[i][0];
     }
 
-    // Entry can be compared for equality (by hash) to another entry (by index).
-    // This can be disabled by comparing to self: entryEqualToOtherEntryIndex[i] = i
-    signal input entryEqualToOtherEntryByIndex[MAX_ENTRIES];
+    /*
+     * Virtual entries. These are entries derived from PODs'
+     * cryptographic data, e.g. the signers' public keys.
+     *
+     * This block pipes them in and forms signals to be fed into the
+     * entry constraint, tuple and list membership modules below, as well
+     * as those revealing their hashes where applicable.
+     */
+    
+    // Maximum number of virtual entries.
+    var MAX_VIRTUAL_ENTRIES = MAX_OBJECTS;
+
+    // Virtual entry value hashes deduced from cryptographic data. At
+    // the moment, this consists of hashed object signers' public
+    // keys.
+    signal virtualEntryValueHashes[MAX_VIRTUAL_ENTRIES];
```

Comment:
Combining the virtual entries conceptually like this is clever, and does save inputs thanks to packing like this.
Do you have any thoughts on how to group/organize multiple types of virtual entires in future?  E.g. if we add content ID?  Is it up to the compiler to know how to combine them all into an array in the correct order?
This approach probably could work for constants (non-POD values) too.  E.g. if you want a comparison value to feed into a less-than comparison it could be a virtual entry specified in the proof inputs.
The bit-packing approach will limit us to a total of 256 virtual entries, but I think that's unlikely to be a problem soon.

Reply:
I was thinking we would agree on an order (e.g. public key hashes first, then content IDs) and have the compiler handle this. That way they can be treated identically.